perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,838 @@
+   populate the final result collection with these information.
+
+  Specifically, we are interested in tps numbers reported by each reporting
+  interval, and the summary latency numbers printed at the end of the run in
+  "General Statistics" -> "Response Time".
+
+  Example Sysbench output:
+
+  sysbench 0.5:  multi-threaded system evaluation benchmark
+  <... lots of output we don't care here ...>
+  Threads started!
+
+  [   2s] threads: 16, tps: 526.38, reads: 7446.79, writes: 2105.52, response
+  time: 210.67ms (99%), errors: 0.00, reconnects:  0.00
+  < .... lots of tps output every 2 second, we need all those>
+
+  < ... lots of other output we don't care for now...>
+  General statistics:
+      total time:                          17.0563s
+      total number of events:              10000
+      total time taken by event execution: 272.6053s
+      response time:
+           min:                                 18.31ms
+           avg:                                 27.26ms
+           max:                                313.50ms
+           approx.  99 percentile:              57.15ms
+  < We care about the response time section above, these are latency numbers>
+  < then there are some outputs after this, we don't care either>
+
+  Args:
+    sysbench_output: The output from sysbench.
+    results: The dictionary to store results based on sysbench output.
+    metadata: The metadata to be passed along to the Samples class.
+  """
+  all_tps = []
+  seen_general_statistics = False
+  seen_response_time = False
+
+  response_times = {}
+
+  sysbench_output_io = StringIO.StringIO(sysbench_output)
+  for line in sysbench_output_io.readlines():
+    if re.match('^\[', line):
+      tps = re.findall('tps: (.*?),', line)
+      all_tps.append(float(tps[0]))
+      continue
+
+    if line.startswith('General statistics:'):
+      seen_general_statistics = True
+      continue
+
+    if seen_general_statistics:
+      if re.match('^ +response time:.*', line):
+        seen_response_time = True
+        continue
+
+    if seen_general_statistics and seen_response_time:
+      for token in RESPONSE_TIME_TOKENS:
+        search_string = '.*%s: +(.*)ms' % token
+        if re.findall(search_string, line):
+          response_times[token] = float(re.findall(search_string, line)[0])
+
+  tps_line = ', '.join(map(str, all_tps))
+  # Print all tps data points in the log for reference. And report
+  # percentiles of these tps data in the final result set.
+  logging.info('All TPS numbers: \n %s', tps_line)
+
+  tps_percentile = _PercentileCalculator(all_tps)
+  for percentile in PERCENTILES_LIST:
+    percentile_string = 'p%s' % str(percentile)
+    logging.info('%s tps %f', percentile_string,
+                 tps_percentile[percentile_string])
+    metric_name = ('%s %s') % (SYSBENCH_RESULT_NAME_TPS, percentile_string)
+    results.append(sample.Sample(
+        metric_name,
+        tps_percentile[percentile_string],
+        NA_UNIT,
```

Comment:
I think these are unit-less. "TPS" is the name of the measurement, and the unit of the measurement is simply a number (a counter) which has no unit.